### PR TITLE
Support get_config method and override required tests, add mock data …

### PR DESCRIPTION
**Pull request by anas-shami**
Support get_config method to fetch initial, running and candidate config. 
Add new tests one for testing 'all' and second one to fetch and validate initial and running data as an individual filter.

### CODE REVIEW

**Comment by anas-shami** [comment]
@samerd  @anasbadaha  please review

File: napalm_onyx/onyx_ssh.py [hunk 1]
```diff
@@ -774,3 +774,39 @@ def traceroute(self,
 
     def get_config(self, retrieve='all'):
         raise NotImplementedError("get_config is not supported yet for onyx devices")
```
Comment (anasbadaha) [inline napalm_onyx/onyx_ssh.py]: Please remove this line then add your code here!

**Comment by samerd** [comment]
@anas-shami 
get_config should return the configuration as text not onyx json:
all other vendors are using: py23_compat.text_type(config.text)
please fix

**Comment by anas-shami** [comment]
oh, its ok I'll use show commands, but in tests we have expected json file ! so I must return json for this method.

File: napalm_onyx/onyx_ssh.py [hunk 2]
```diff
@@ -774,3 +774,39 @@ def traceroute(self,
...
+                    ]
+                },
+            'startup': '',
+            'candidate': ''
+        }
+        """
+        config = {"startup": "", "running": "", "candidate": ""}  # default values
+
+        if retrieve in ('running', 'all'):
+            command = 'show running-config | json-print'
```
Comment (samerd) [inline napalm_onyx/onyx_ssh.py]:
```python
command = 'show running-config'
result = self._send_command(command)
config['running'] = py23_compat.text_type(result )
```

**Review by samerd — CHANGES_REQUESTED** [review]

**Comment by anas-shami** [comment]
I don't make a pull --rebase here this fails on py3.4  @samerd

**Comment by anas-shami** [comment]
@samerd  could you please approve that so I can continue with other functions?

**Review by samerd — COMMENTED** [review]
Thanks @anas-shami 
seems great,
However it is still failing.
I suggest to work with a new updated clean branch

**Pull request closed without merging**